### PR TITLE
Implement {DataFrame, Series}.empty

### PR DESCRIPTION
**Pull request by Fernadoo**
## What do these changes do?

I re-implemented {DataFrame, Series}.empty and add related test cases.

## Related issue number

Issue #1830 should be resolved

### CODE REVIEW

**Comment by Fernadoo** [comment]
@qinxuye Not so sure if I correctly raise the `ValueError` as you mean :D

**Comment by Fernadoo** [comment]
Now there are 2 blank lines before `test_empty()` and 1 blank line at EoF

**Comment by qinxuye** [comment]
> @qinxuye Not so sure if I correctly raise the `ValueError` as you mean :D

LGTM

**Comment by qinxuye** [comment]
> Now there are 2 blank lines before `test_empty()` and 1 blank line at EoF

Just track the flake8 verification in CI to make sure it's passed.

File: mars/dataframe/core.py [hunk 1]
```diff
@@ -1021,6 +1021,13 @@ def index(self):
     def axes(self):
         return [self.index]
 
+    @property
+    def empty(self):
+        shape = getattr(self, '_shape')
+        if np.any(np.isnan(shape)):
+            raise ValueError('Tileable object must be executed first')
```
Comment (qinxuye) [inline mars/dataframe/core.py:1028]: Please add a unittest for Series as well.

Comment (Fernadoo) [reply mars/dataframe/core.py:1028]: By `unittest`, you mean something like `a[a>0]` where `a` is a randomly initialized Series, like what I did for dataframe to make it raise ValueError?

Comment (qinxuye) [reply mars/dataframe/core.py:1028]: Right, the code I mentioned is not covered by any tests.

Comment (Fernadoo) [reply mars/dataframe/core.py:1028]: Hi @qinxuye, I have rebased to the latest commit and also added the unitest for Series data. Plz have a look ~

Comment (qinxuye) [reply mars/dataframe/core.py:1028]: Thanks for you contribution, so far it looks good, I will review it after CI finished.

**Comment by qinxuye** [comment]
Flake8 failed, see https://github.com/mars-project/mars/pull/2155/checks?check_run_id=2829247028 .

and refer to https://www.flake8rules.com/rules/W391.html to see why.

**Comment by qinxuye** [comment]
Little tip, you can run flake8 verification locally, copy the command here: https://github.com/mars-project/mars/blob/fcc50d4530561bb55105baf8aa34244104ad10f3/.github/workflows/core-ci.yml#L94

when the working dir is mars directory.

**Comment by Fernadoo** [comment]
I guess now it should be okay
截屏2021-06-15 下午8 39 27
This tricky bug is due to the inconsistency of treating blank lines by sublime and vim

**Comment by Fernadoo** [comment]
Seems like [this test](https://github.com/mars-project/mars/pull/2155/checks?check_run_id=2829596322) has run for a long time. Is this wired?

**Comment by Fernadoo** [comment]
Just wonder is this failure https://github.com/mars-project/mars/pull/2155/checks?check_run_id=2830164180 due to my implementation?

**Comment by Fernadoo** [comment]
> Just wonder is this failure https://github.com/mars-project/mars/pull/2155/checks?check_run_id=2830164180 due to my implementation?

Seems that similar error also happened here https://github.com/mars-project/mars/runs/2784234334

**Comment by qinxuye** [comment]
> Just wonder is this failure https://github.com/mars-project/mars/pull/2155/checks?check_run_id=2830164180 due to my implementation?

It's irrelevant, the related people is working on the possible failure.

**Review by qinxuye — APPROVED** [review]
LGTM

**Comment by Fernadoo** [comment]
> > Just wonder is this failure https://github.com/mars-project/mars/pull/2155/checks?check_run_id=2830164180 due to my implementation?
> 
> It's irrelevant, the related people is working on the possible failure.

Thxxxxxx so much

**Review by hekaisheng — APPROVED** [review]
LGTM

**Comment by qinxuye** [comment]
Thanks for excellent work, looking forward to seeing more contributions from you.

**Comment by Fernadoo** [comment]
Really thank you guys @qinxuye @hekaisheng for helping me for such a long time. It's been an incredible experience to fit into this kind of open source project!!

**Pull request merged**